patroni/api.py

```diff
@@ -197,7 +197,7 @@ def _read_json_content(self, body_is_optional=False):
     def do_PATCH_config(self):
         request = self._read_json_content()
         if request:
-            cluster = self.server.patroni.dcs.get_cluster()
+            cluster = self.server.patroni.dcs.get_cluster(True)
```

Comment:
If it is meant to be [used with Zookeeper](https://github.com/zalando/patroni/blob/d7f579ee61822cd0fc181a0783c2b920f902734a/patroni/dcs/__init__.py#L657-L662), it is worth an explicit check the DCS is indeed Zookeper.

Reply:
Caller doesn't know anything about DCS implementation.
It is sort of possible to try to load the very specific DCS and use isinstance for comparison, but loading unnecessary modules increases memory footprint.